compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt

```diff
@@ -19,9 +19,45 @@ import org.jetbrains.kotlin.fir.declarations.impl.FirDefaultPropertyAccessor
 import org.jetbrains.kotlin.fir.types.FirImplicitTypeRef
 import org.jetbrains.kotlin.lexer.KtTokens
 
+internal fun isInsideExpectClass(containingClass: FirRegularClass, context: CheckerContext): Boolean =
+    isInsideSpecificClass(containingClass, context) { klass -> klass.isExpect }
+
+internal fun isInsideExternalClass(containingClass: FirRegularClass, context: CheckerContext): Boolean =
+    isInsideSpecificClass(containingClass, context) { klass -> klass.isExternal }
+
 // Note that the class that contains the currently visiting declaration will *not* be in the context's containing declarations *yet*.
-internal fun isInsideExpectClass(containingDeclaration: FirRegularClass, context: CheckerContext): Boolean =
-    containingDeclaration.isExpect || context.containingDeclarations.asReversed().any { it is FirRegularClass && it.isExpect }
+private inline fun isInsideSpecificClass(
+    containingClass: FirRegularClass,
+    context: CheckerContext,
+    specificStatus: (FirRegularClass) -> Boolean
```

Comment:
Minor: Suggest naming this `containingClassPredicate` or just `predicate`

Reply:
Edited as per the suggestion, thanks.